### DISCUSSION TITLE
`magick convert` behaves differently from `magick` (4 times slower)

### DISCUSSION THREAD

**Discussion by tobwen** [ImageMagick/ImageMagick#7485] [Help] [answered]
### ImageMagick version

7.1.1-35 Q16-HDRI 1bfce2a62

### Operating system, version and so on

Debian Buster, AMD64

### Description

Before reporting it as a bug, I want to have a talk. I have noticed that there are big differences between `magick convert` and `magick` in the runtime.

### `magick convert`
``` sh
$ time magick convert -monitor test.png -fx "(u.hue>0)? 255 : u" -threshold 75% -compress Fax test.tif
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"

load image[test.png]: 2338 of 2339, 100% complete
list
FxNew/Image[test.png]: 2338 of 2339, 100% complete
contrast-stretch image[test.tif]: 2338 of 2339, 100% complete
classify image colors[test.tif]: 2338 of 2339, 100% complete
dither image colors[test.tif]: 3866366 of 3866367, 100% complete
save image[test.tif]: 2338 of 2339, 100% complete

real    0m5.297s
user    0m14.281s
sys     0m0.410s
```

### `magick`
``` sh
$ time magick -monitor test.png -fx "(u.hue>0)? 255 : u" -threshold 75% -compress Fax test.tif
load image[test.png]: 2338 of 2339, 100% complete
FxNew/Image[test.png]: 2338 of 2339, 100% complete
contrast-stretch image[test.tif]: 2338 of 2339, 100% complete
threshold image[test.tif]: 2338 of 2339, 100% completeomplete
classify image colors[test.tif]: 2338 of 2339, 100% complete
dither image colors[test.tif]: 3866366 of 3866367, 100% complete
save image[test.tif]: 2338 of 2339, 100% complete

real    0m20.865s
user    0m19.312s
sys     0m4.489s
```

### hint
While `magick convert` skips the threshold, `magick` takes way longer in dither. Perhaps some kind of compatibility mode, where `magick convert` shows the old and bad behavior?

**Comment by fmw42** [comment, marked as answer]
IM 7 magick was a major update to IM 6 convert.  The command line syntax got more strict. Many new features were added. Magick convert is somewhere between and is the old convert syntax called from IM 7.  Expect many differences.

See

https://imagemagick.org/script/porting.php#cli

**Comment by DadeMurphy72** [comment]
If I'm not mistaken, Q16 can slow down processing as well. Have you tried Q8?
